bundles/statistics/statsgrid2016/view/SearchFlyout.js

```diff
@@ -66,28 +66,34 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.SearchFlyout', function (t
     getNewSearchElement: function () {
         var me = this;
         var container = jQuery('<div></div>');
-
         var selectionComponent = Oskari.clazz.create('Oskari.statistics.statsgrid.IndicatorSelection', me.instance, me.sandbox);
         container.append(selectionComponent.getPanelContent());
 
+        var buttonContainer = jQuery('<div id="statistics-search-flyout-button-container"></div>');
+        container.append(buttonContainer);
+
         var btn = Oskari.clazz.create('Oskari.userinterface.component.Button');
         btn.addClass('margintopLarge');
         btn.setPrimary(true);
         btn.setTitle(this.loc('panels.newSearch.addButtonTitle'));
         btn.setEnabled(false);
-        btn.insertTo(container);
+        btn.insertTo(buttonContainer);
 
         btn.setHandler(function (event) {
             event.stopPropagation();
-            me.setSpinner(selectionComponent.spinner);
+            const progressSpinner = Oskari.clazz.create('Oskari.userinterface.component.ProgressSpinner');
+            progressSpinner.insertTo(jQuery('#statistics-search-flyout-button-container'));
```

Comment:
You can just pass the buttonContainer variable. Also remove the id as we try to use as little those as possible.

Reply:
Actually you can do the setup outside of the button handler and only start the spinner in the handler.

Reply:
In latest commit, spinner setup logic is moved from button event handler function. Also id attribute is removed from button container div.